vita3k/gui/src/content_manager.cpp

```diff
@@ -83,94 +85,119 @@ struct SaveData {
+static std::map<std::string, std::string> space;
+static std::map<std::string, bool> contents_selected, finished_get_size;
+static size_t apps_list_size, save_data_list_size;
 
 static void get_save_data_list(GuiState &gui, HostState &host) {
     host.io.user_id = fmt::format("{:0>2d}", host.cfg.user_id);
     save_data_list.clear();
+    save_data_list_size = 0;
+
+    std::thread get_save_data_size([&gui, &host]() {
+        fs::path SAVE_PATH{ fs::path{ host.pref_path } / "ux0/user" / host.io.user_id / "savedata" };
+        if (!fs::exists(SAVE_PATH))
+            return;
+
+        for (const auto &save : fs::directory_iterator(SAVE_PATH)) {
+            const auto title_id = save.path().stem().generic_string();
+            if (fs::is_directory(save.path()) && !fs::is_empty(save.path()) && (get_app_index(gui, title_id) != gui.app_selector.user_apps.end())) {
+                const auto last_writed = fs::last_write_time(save);
+                const auto updated = std::localtime(&last_writed);
+                const auto date = fmt::format("{}/{}/{} {:0>2d}:{:0>2d}", updated->tm_mday, updated->tm_mon + 1, updated->tm_year + 1900, updated->tm_hour, updated->tm_min);
+
+                size_t save_data_size = 0;
+                for (const auto &save_path : fs::recursive_directory_iterator(save)) {
+                    if (fs::is_regular_file(save_path.path()))
+                        save_data_size += fs::file_size(save_path.path());
+                }
+                save_data_list_size += save_data_size;
 
-    fs::path SAVE_PATH{ fs::path{ host.pref_path } / "ux0/user" / host.io.user_id / "savedata" };
-    if (!fs::exists(SAVE_PATH))
-        return;
-
-    for (const auto &save : fs::directory_iterator(SAVE_PATH)) {
-        const auto title_id = save.path().stem().generic_string();
-        if (fs::is_directory(save.path()) && !fs::is_empty(save.path()) && get_app_index(gui, title_id) != gui.app_selector.user_apps.end()) {
-            const auto last_writed = fs::last_write_time(save);
-            const auto updated = std::localtime(&last_writed);
-            const auto date = fmt::format("{}/{}/{} {:0>2d}:{:0>2d}", updated->tm_mday, updated->tm_mon + 1, updated->tm_year + 1900, updated->tm_hour, updated->tm_min);
-
-            size_t size = 0;
-            for (const auto &save_path : fs::recursive_directory_iterator(save)) {
-                if (fs::is_regular_file(save_path.path()))
-                    size += fs::file_size(save_path.path());
+                save_data_list.push_back({ get_app_index(gui, title_id)->title, title_id, save_data_size, date });
             }
+        }
+        space["savedata"] = save_data_list_size ? get_unit_size(save_data_list_size) : "-";
+        finished_get_size["savedata"] = true;
+        std::sort(save_data_list.begin(), save_data_list.end(), [](const SaveData &sa, const SaveData &sb) {
+            return sa.title < sb.title;
+        });
+    });
+    get_save_data_size.detach();
```

Comment:
You cant use detach here. We need to properly clean the states for this case.

Reply:
I have no understund code works fine here 🤔

Reply:
It can break randomly.

Reply:
Fir niw after litteraly 100 lunch never get any issue here no really understund what you want exatcly 🤔

Reply:
Study race condition. Otherwise, you wont understand.

Reply:
Oh btw for detach, its related to exitting in the middle. The thread can be spawned multiple tunes.

Reply:
Which can mess up the things.

Reply:
No sur understund here, it run function one time and after exit it so it no can run again this detach 🤔

Reply:
If it runs once, then the size is not calculated again unless restarting the emu?

Reply:
Run again so close content manager, etbre open it re calcule it yes, is for that value is cleaned before

Reply:
Thread will not exit even if you close content manager.

Reply:
Yes i know hmm i need thinks, you know how kill thread if run it is not finissed in moment  close content manager?

Reply:
i have founded and fix that, now it stop thread in moment exit content manager

Reply:
That's also not thread safe and hacky. I'll just send a commit that fixes the issues.

Reply:
oki, i have re force push